kernel/kernel-api/src/main/java/io/delta/kernel/CheckpointAlreadyExistsException.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel;
+
+import static java.lang.String.format;
+
+import io.delta.kernel.annotation.Evolving;
+
+/**
+ * Thrown when trying to create a checkpoint at version {@code v}, but there already exists
+ * a checkpoint at version {@code v}.
+ *
+ * @since 3.2.0
+ */
+@Evolving
+public class CheckpointAlreadyExistsException extends IllegalArgumentException {
```

Comment:
@allisonport-db Let me know if I should remove this and as part of the new exception framework.